compiler/modules/front/codeGen.bal

```diff
@@ -801,12 +801,15 @@ function codeGenAssignToVar(CodeGenContext cx, bir:BasicBlock startBlock, Enviro
         unnarrowedReg = unnarrowedBinding.reg;
         assignments = [ unnarrowedReg.number ];
     }
-    s:Expr foldedExpr = check cx.foldExpr(env, expr, unnarrowedReg.semType);
-    var { result: operand, block: nextBlock } = check codeGenExpr(cx, startBlock, env, foldedExpr);
+    return { block: check codeGenAssign(cx, env, startBlock, unnarrowedReg, expr, unnarrowedReg.semType), assignments };
```

Comment:
Better to split this over multiple lines.

Reply:
Fixed